### PR TITLE
VRS integration into genomicVariations  default schema

### PR DESCRIPTION
**Pull request by mbaudis**
This is an attempt to allow the use of VRS variation classes, while also keeping an updated version of the current Beacon schema (using VRS location, which was agreed upon) as an option. This proposal tries to address feedback from the PRC process.

All the additional properties beyond the definition of the variant itself (annotations, caseLevel data etc.) are kept in place.

The change here introduces the root level `variation` property but other options could be possible.

### CODE REVIEW

**Comment by mrueda** [comment]
Hi Michael, just getting into this now...there are terms in VRS.location" that are easily transferable (e.g., `interval` or `type`), but what about others such as [sequence_id](https://vrs.ga4gh.org/en/stable/impl-guide/computed_identifiers.html)?

**Comment by mbaudis** [comment]
@mrueda VRS does not require computed identifiers, only that CURIEs are being used. See e.g. https://vrs.ga4gh.org/en/stable/impl-guide/example.html . So `sequenceId` should usually be a refseq id, which then also obviates the need for the definition of a genome edition (`"sequenceId": "refseq:NC_000001.10"` for chr1, GRCh37).

**Comment by mbaudis** [comment]
As discussed today (and in several conversations, also with @ahwagner) this seems to be the best way to go ATM. I'm +1 on merge, but leave the step to @jrambla / @MrRobb .

**Comment by Tom-Shorter** [comment]
I believe that this is a good change that has to be made in one form or another, I'll wait to see what @jrambla says before approving this PR though but I can't see any problems with it currently.

I echo the sentiments made by @mrueda about the mapping of current beacon terms onto VRS. Almost all of these are 1-1 mappings but as pointed out sequence_id is created by a combination of current fields.

I think it's worth taking some time to make a document with all VRS fields and then how these are created from the current beacon fields. Not only will this ensure we are all happy that everything that beacon offers can be handled by VRS it also makes the life of implementers, who need to make the changes, easier as they wont have to do the mappings themselves. This document might also be useful as part of the re-submission to show how Beacon has been changed to better align with external standards.



~~~~ the below is related but also probably a separate issue~~~~

Do we also need to think about the beacons which use different names/identifiers for chromosomes? Some people wont store the refseq id of chromosomes, instead just stating they use (chr)1 of GrCH3X which could be turned into a CURIE along the lines of `chr:1` or `chromosome:1` or `grch37:1`. This causes no problems when querying a single beacon instance from a interface designed to query that beacon but as soon as you start querying across multiple beacons this becomes an issue.

I would recommend the inclusion of a statement such as:

> Beacon recommends the use of refseq for identifying chromosomes.

This problem will likely appear in more places than just chromosome.

Beacon has done something before, in beacon v1 genomic positions were denoted in 0 based half closed (or something similar) notation and this was stated in the documentation so implementers were aware they would need to do some form of conversion to fit the query to their data.

If anyone agrees with this then we should split this off into another issue but if it's not of concern then I'm happy to leave it here.

**Comment by mbaudis** [comment]
@Tom-Shorter Thanks for the elaboration - indeed, this is a "we have to be more prescriptive due to alignment with external standards" issue. But this is on the model side, so supporting clean data delivery... I would argue that we should recommend & document the CURIE-format refseq-id or alternative ga4gh/VRS identifier use.

(The alternative would be to keep the Position object in its previous form for the LegacyVariation while keeping the suggested  change to a `oneOf` root `variation` => It probably would have made sense to keep the 2 changes for separate discussion.)

File: BEACON-V2/genomicVariations/examples/genomicVariant-MIN-VRS-example.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "$schema": "../defaultSchema.json",
+  "variantInternalId": "GRCh37-1-55505652-G-A",
+  "variation":{
+    "state": {
+      "sequence": "A",
+      "type": "SequenceState"
```
Comment (Tom-Shorter) [inline BEACON-V2/genomicVariations/examples/genomicVariant-MIN-VRS-example.json:7]: This is deprecated in VRS, `LiteralSequenceExpression` should be used instead from what I can tell.

Comment (mbaudis) [reply BEACON-V2/genomicVariations/examples/genomicVariant-MIN-VRS-example.json:7]: As I understand the schema this is correct here: `state` is a `SequenceExpression` (or alternatively a `SequenceState`) which can be a `LiteralSequenceExpression` which has the properties `type` and `sequence`. So`state` collapses to those 2 properties.

Comment (Tom-Shorter) [reply BEACON-V2/genomicVariations/examples/genomicVariant-MIN-VRS-example.json:7]: That's what I thought at first as well but then in the [vrs.yaml file](https://github.com/ga4gh/vrs/blob/main/schema/vrs.yaml) SequenceState is set to deprecated: true (line 619-20).

**Review by Tom-Shorter — APPROVED** [review]

**Review by MrRobb — APPROVED** [review]

File: BEACON-V2/genomicVariations/defaultSchema.json [hunk 2]
```diff
@@ -6,37 +6,21 @@
...
     "variantInternalId",
-    "variantType",
-    "alternateBases",
-    "position"
+    "variation"
   ],
   "properties": {
+    "variation": {
+      "oneOf": [
+        {"$ref": "https://raw.githubusercontent.com/ga4gh/vrs/1.2.0/schema/vrs.json#/definitions/MolecularVariation" },
```
Comment (ahwagner) [inline BEACON-V2/genomicVariations/defaultSchema.json:14]: I suggest using 1.2 instead of 1.2.0; this will enable use of any patch-level fixes

**Review by ahwagner — APPROVED** [review]
This is a reasonable compromise solution that allows the use of both legacy and VRS variation. My only comment is to reference the VRS schema from the `1.2` branch instead of the `1.2.0` tag.

File: BEACON-V2/genomicVariations/defaultSchema.json (hunk 2 above)
Comment (mbaudis) [reply BEACON-V2/genomicVariations/defaultSchema.json]: Thanks - modified with https://github.com/ga4gh-beacon/beacon-v2-Models/pull/72/commits/8a0fa5ed3efa320584183bdfb2a5ab39e029a646

**Comment by jrambla** [comment]
+1 to the mapping documentation wisely suggested by Tom. 
Part of the conversation above lies in the fundamental design decision on:
As Beacon is a discovery but also an entry tool for GA4GH and data sharing, to which end of the "easy vs standard" (or "easy entry vs. precise discovery") continuum it must be shifting.
So far, I'd favored "easy" as I believe that using powerful standards comes naturally once you have started to share.
Of course that is having a cost on the fostering (enforcing?) standards end of the continuum.
Beacon has plenty of such "debates" as it is in a centric position.
They are welcome for that reason ;-)

**Comment by jrambla** [comment]
@mbaudis, naive question: `SequenceState `is to be deprecated in VRS2.0.
Have we discussed that already?

**Pull request merged**